src/libostree/ostree-repo-commit.c

```diff
@@ -734,6 +734,17 @@ write_content_object (OstreeRepo         *self,
     }
   else
     {
+      /* Update size metadata if configured */
```

Comment:
Can you elaborate on how this makes a difference? Aren't `phys_object_is_symlink` and `object_file_type == G_FILE_TYPE_REGULAR` already mutually exclusive?